01_Data/src/layers/sequelize/repository/LocalAuthList.ts

```diff
@@ -0,0 +1,243 @@
+        this.sendLocalList = sendLocalList ? sendLocalList : new SequelizeRepository<SendLocalList>(config, SendLocalList.MODEL_NAME, logger, sequelizeInstance);
+    }
+
+    async createSendLocalListFromStationIdAndRequest(stationId: string, sendLocalListRequest: SendLocalListRequest): Promise<SendLocalList> {
+        return this.createSendLocalList(stationId, sendLocalListRequest.updateType, sendLocalListRequest.versionNumber, sendLocalListRequest.localAuthorizationList ?? undefined);
+    }
+
+    async createSendLocalList(stationId: string, updateType: UpdateEnumType, versionNumber?: number, localAuthorizationList?: AuthorizationData[]): Promise<SendLocalList> {
+        if (versionNumber) {
+            if (versionNumber <= 0) {
+                throw new Error(`Version number ${versionNumber} must be greater than 0, see D01.FR.18`)
+            }
+            const localListVersion = await LocalListVersion.findOne({ where: { stationId }, include: [LocalListAuthorization] });
+            if (localListVersion && localListVersion.versionNumber >= versionNumber) {
+                throw new Error(`Current LocalListVersion for ${stationId} is ${localListVersion.versionNumber}, cannot send LocalListVersion ${versionNumber} (version number must be higher)`);
+            }
+        } else {
+            versionNumber = await this.getNextVersionNumberForStation(stationId);
+        }
+        if (localAuthorizationList && localAuthorizationList.length > 1) { // Check for duplicate authorizations
+            const idTokens = localAuthorizationList.map(auth => auth.idToken.idToken + auth.idToken.type);
+            if (new Set(idTokens).size !== idTokens.length) {
+                throw new Error(`Duplicated idToken in SendLocalListRequest ${JSON.stringify(idTokens)}`);
+            }
+        }
+        const sendLocalList = await SendLocalList.create({
+            stationId,
+            correlationId: uuidv4(),
+            versionNumber,
+            updateType,
+        });
+        localAuthorizationList?.map(async (authData) => {
+            const auth = await Authorization.findOne({
+                include: [
+                    {
+                        model: IdToken,
+                        where: {
+                            idToken: authData.idToken.idToken,
+                            type: authData.idToken.type,
+                        },
+                    },
+                    {
+                        model: IdTokenInfo,
+                        include: [IdToken],
+                    },
+                ],
+            });
+            if (!auth) {
+                throw new Error(`Authorization not found for ${JSON.stringify(authData)}, invalid SendLocalListRequest (create necessary Authorizations first)`);
```

Comment:
I think this throw may be unhandled in the map promise, worth a double check but I think here you are throwing the error within a dynamic async function that you are passing into .map. The map will end up returning a list of promises which will not be executed in this case until further down when there is an await present on line 111. And so there is no explicit handling of the promises to ensure that any errors can be caught which I think what may cause the throws to result in unhandled rejections, and those errors may go unnoticed

Reply:
good catch, fixed!